Feat/intro (#16)
computer-aided-biotech/better-cb: content/02.introduction.md

Scientific computing requires following specific practices to enable shareable, reproducible and sustainable outputs that stand the test of time. Computing-heavy disciplines such as software engineering and data science have already adopted practices addressing the need for collaboration, visualization, project management, and strengthening of online communities. However, as a highly interdisciplinary and evolving field, computational biology has yet to acquire a set of universal "best practices". As most computational biologists come from diverse backgrounds and oftentimes lack formal computational training, the absence of guidelines can lead to disparate and unsustainable practices that hinder reproducibility and collaboration, and slow down biomedical and biotechnological research.

> Over the last decade, some works have focused on delivering advice directed to bench scientists starting in either scientific computing [@doi:10.1371/ journal.pcbi.1005510, doi:10.1371/journal.pbio.1001745, @doi:10.1093/gigascience/giz054,] or computational biology [@doi:10.1038/nbt.2740]. The advice encompasses a wide range of topics ranging from programming to project organization to manuscript writing. Other works have adopted a different approach, fixating in one powerful tool and diving deeper into its scope and applications, such as the software development and version control cloud service GitHub [@doi:10.1371/journal.pcbi.1004947] and the web-application Jupyter Notebooks [@arXiv:1810.08055v1]. Similarly, recent works have chosen to comprehensively address one specific need in computational biology such as workflow automation [@doi:10.1101/2020.06.30.178673] and software libraries development [@doi:10.1016/j.cels.2017.08.003]. Although this advice proves immensely helpful, several aspects of the computational biology "journey" remain uncovered. Specifically, there is a lack of a clear roadmap regarding best practices from fundamental to advanced topics, in addition to practical examples tackling the complexities of this kind of research. 

Reviewer: Some comments on this new paragraph:
* How about changing "some works have focused on delivering advice" to something more active e.g. "several researchers have published advice"?
* The first reference has a space, creating an error in the output html.
* Whenever you list many references, you should separate them with semicolon, not comma, otherwise manubot will not parse it properly.
* I think "software library development" reads better than "software libraries development".

All suggestions are included below:

```suggestion
Over the last decade, several researchers have published advice directed to bench scientists starting in either scientific computing [@doi:10.1371/journal.pcbi.1005510; doi:10.1371/journal.pbio.1001745; @doi:10.1093/gigascience/giz054,] or computational biology [@doi:10.1038/nbt.2740]. The advice encompasses a wide range of topics ranging from programming to project organization to manuscript writing. Other works have adopted a different approach, fixating in one powerful tool and diving deeper into its scope and applications, such as the software development and version control cloud service GitHub [@doi:10.1371/journal.pcbi.1004947] and the web-application Jupyter Notebooks [@arXiv:1810.08055v1]. Similarly, recent works have chosen to comprehensively address one specific need in computational biology such as workflow automation [@doi:10.1101/2020.06.30.178673] and software library development [@doi:10.1016/j.cels.2017.08.003]. Although this advice proves immensely helpful, several aspects of the computational biology "journey" remain uncovered. Specifically, there is a lack of a clear roadmap regarding best practices from fundamental to advanced topics, in addition to practical examples tackling the complexities of this kind of research. 
```

My last comment is that the last sentence of the paragraph sounds a bit broad, but let's revisit it once we have a more developed manuscript, as we will be able to better exemplify what makes this review different.

Author: I agree! we should revisit the last paragraph of the introduction after the manuscript is finished.

> We premise that best practices in computational biology lie within a continuum that traverses three "levels": the individual’s internal practices, the collaborative practices of a team, and the practices that allow a broader scientific community to access and engage over time with the tools and resources developed. Each one of these levels has a different set of needs and challenges. Here, we compiled a selected list of relevant practices and related tools advisable for each one these levels, emphasizing their time and place in a computational biology research project. We illustrated the utility of these practices in three case studies covering the broader spectrum of computational biology research.

Reviewer: ```suggestion
We premise that best practices in computational biology lie within a continuum that traverses three "levels": the individual’s internal practices, the collaborative practices of a team, and the practices that allow a broader scientific community to access and engage over time with the developed research. Each one of these levels has a different set of needs and challenges. Here, we compile a selected list of relevant practices and related tools advisable for each one of these levels, emphasizing their time and place in a computational biology research project. Finally, we illustrate the utility of these practices in three case studies covering the broader spectrum of computational biology research.
```

## Introduction

> Since Margaret Dayhoff pioneered the field of bioinformatics back in the decade of the sixties, the application of computational tools to the field of biology has vastly grown in scope and impact. Nowadays, biotechnological and biomedical research are routinely fed by the insights arising from novel computational approaches, machine learning algorithms and mathematical models. The ever increasing amount of biological data and the exponential growth in computing power will amplify this trend in the years to come.

Reviewer: ```suggestion
Since Margaret Dayhoff pioneered the field of bioinformatics back in the sixties, the application of computational tools to the field of biology has vastly grown in scope and impact. Nowadays, biotechnological and biomedical research are routinely fed by the insights arising from novel computational approaches, machine learning algorithms and mathematical models. The ever increasing amount of biological data and the exponential growth in computing power will amplify this trend in the years to come.
```

> The use of computers to address biological matters encompasses a wide array of applications usually grouped under the terms of computational biology and bioinformatics. Although distinct definitions have been delineated for each one, here we will consider “computational biology” as the umbrella term for any application that involves the intersection of computing and modern biological data. As such, a computational biologist can be a bioinformatician, a statistician, a mathematical modeler, a data analyst, a data engineer, a software developer, and many others. In praxis, the modern computational biologist will be a “scientist of many hats”, taking on several of the duties listed above. But first and foremost, we will consider a computational biologist as a scientist whose ultimate goal is answering a biological question or addressing a need in the life sciences.

Reviewer: ```suggestion
The use of computers to address biological matters encompasses a wide array of applications usually grouped under the terms of "computational biology" and "bioinformatics". Although distinct definitions have been delineated for each one, here we will consider “computational biology” as the umbrella term for any application that involves the intersection of computing and modern biological data [REF]. As such, a computational biologist can be a bioinformatician, a statistician, a mathematical modeler, a data analyst, a data engineer, a software developer, and many others. In praxis, the modern computational biologist will be a “scientist of many hats”, taking on several of the duties listed above. But first and foremost, we will consider a computational biologist as a scientist whose ultimate goal is answering a biological question or addressing a need in the life sciences.
```

~- [ ] We should define what we mean by bioinformatics somewhere~
- [x] We should add a reference for the definition of comp. bio.

Author: Yes! The problem is that there are conflicting definitions about bioinformatics and computational biology.
- Here is the NIH working definition of these concept which deeply confuses me: http://www.binf.gmu.edu/jafri/math6390-bioinformatics/workingdef.pdf
- Here is a whole... article? that defines bioinformatics in a more specific way: https://pubmed.ncbi.nlm.nih.gov/11552348/
- Here is another confusing definition: https://www.medicaltechnologyschools.com/biotechnology/bioinformatics-vs-computational-biology
- And here is Nature's definition which basically stick them together: https://www.nature.com/subjects/computational-biology-and-bioinformatics#:~:text=Definition,predictions%20or%20discover%20new%20biology.
We can discuss this in our in-person meeting, because it's quite convoluted.

Reviewer: Computational Biology has now been referenced, and as discussed Bioinformatics will not be defined :)

> Computing requires following specific practices to enable shareable, reproducible and sustainable outputs that stand the test of time. Computing-heavy disciplines such as software development and data science have already adopted multiple practices addressing the need for project management, collaboration, visualization and strengthening of online communities. As a highly interdisciplinary and growing field, computational biology has yet to acquire a set of universal "best practices". Moreover, as most computational biologists come from diverse backgrounds, they are usually let on their own determining the best ways to conduct their research.

Reviewer: - [x] The transition from the previous paragraph to this one feels sudden. We should think of a way of better connecting them, e.g. ending the previous one with an emphasis on using tools we can trust.
- [x] _"[...] as most computational biologists come from diverse backgrounds, they are usually let on their own determining the best ways to conduct their research"_ don't see how those 2 ideas connect, maybe we should clarify further.

Author: Totally agree that the transition is not smooth. I will revisit it.
Regarding the second point, what I wanted to say is that there is no formal education as "bioinformatician" or "computational biologist" that includes a standard set of practices taught at University, for example. Therefore, computational biologist are usually let on their own learning best practices to do research. Would you agree with that?

* Where is the community in terms of best practices / Cite literature in the topic

> We premise that best practices in computational biology lie within a continuum that traverses the individual’s internal practices, the collaborative practices of a team, and the practices that allow a broader scientific community to access, apply and engage over time with the tools and approaches developed. Each “level” has a different set of needs and challenges. 

Reviewer: ```suggestion
We premise that best practices in computational biology lie within a continuum that traverses three "levels": the individual’s internal practices, the collaborative practices of a team, and the practices that allow a broader scientific community to access, apply and engage over time with the tools and approaches developed. Each of these levels has a different set of needs and challenges. 
```

I really like this sentence!
